tests/ui/consts/const-eval/heap/alloc_intrinsic_untyped.rs

```diff
@@ -7,5 +7,6 @@ use std::intrinsics;
 
 const BAR: *mut i32 = unsafe { intrinsics::const_allocate(4, 4) as *mut i32 };
 //~^ error: mutable pointer in final value of constant
+//~| error: encountered `const_allocate` pointer in final value that was not leaked
```

Comment:
We also need a test that only errors with this error message and looks otherwise benign